### ISSUE TITLE
[Bug] [iOS] CollectionView incorrectly performs layout on iOS

### ISSUE THREAD

**Issue by DuncWatts** [xamarin/Xamarin.Forms#11607]
### Description
CollectionView shows incorrect gaps between items. The controls snap into the correct location when scrolling begins.

### Steps to Reproduce

1. Add a CollectionView with a template that contains margins and text that wraps
2. View control

### Expected Behavior
Controls are laid out correctly.

### Actual Behavior
Incorrect spacing between items. The controls snap into the correct location when you start to scroll the view.

### Basic Information

- Version with issue:
- Last known good version: None
- IDE: VS2019
- Nuget Packages: Xamarin.Forms 4.7.0.1179
- Affected Devices: Tested with iPhone 6S

### Screenshots

When the page is loaded:
![IMG_0009](https://user-images.githubusercontent.com/23555469/88935321-860b8880-d279-11ea-83f4-d199f91a1ce6.PNG)

As soon as a scroll starts to happen:
![IMG_0010](https://user-images.githubusercontent.com/23555469/88935442-a9363800-d279-11ea-9715-cd0f977bb255.PNG)

### Reproduction Link

https://github.com/DuncWatts/CollectionViewRepro

### Workaround

None

**Comment by AntonIOIOIO** [comment]
This happens when views have different heights inside the collectionview. This is not ok as we have a duplicate issue #7699 that is almost a year old. The collectionview should be in preview until this is fixed. New people are gonna use the collectionview and run into this bug and be really confused. @samhouts @

**Comment by Tommigun1980** [comment]
> This happens when views have different heights inside the collectionview. This is not ok as we have a duplicate issue #7699 that is almost a year old. The collectionview should be in preview until this is fixed. New people are gonna use the collectionview and run into this bug and be really confused. @samhouts @davidortinau @jamesmontemagno

I have given up on the collection view and only use BindableLayouts with a scroll view. It seems to be the most consistent way of drawing content of arbitrary size that works to some degree.

**Comment by jefffhaynes** [comment]
@AntonIOIOIO agreed.  No idea why Xamarin can't stop putting extremely buggy controls into release.  Total lack of discipline in the team as far as I can tell.

**Comment by DuncWatts** [comment]
Ah so it's a caching fault with multiple items sharing heights.

I've modified my example with a spacer that gives a random height between 0.0 and 0.1px for each item and that seems to be a workaround for now.

**Comment by PureWeen** [comment]
@DuncWatts what iOS version are you running on?

I tried on a few iPhone 6s's and I'm not seeing that first layout


![image](https://user-images.githubusercontent.com/5375137/89591314-7bb83400-d80f-11ea-90e5-afbd3054546d.png)

**Comment by DuncWatts** [comment]
@PureWeen I'm using a physical device running iOS 13.6

**Comment by rmarinho** [comment]
@duncanmak what device?

**Comment by rmarinho** [comment]
Ok got it to reproduce on the SE 2nd generation.

When the repo loads: 
![Simulator Screen Shot - iPhone SE (2nd generation) - 2020-08-12 at 11 10 54](https://user-images.githubusercontent.com/1235097/90003829-8f8aec80-dc8c-11ea-99d0-382be41e6f90.png)

If I try to scroll it fixes the issue 
![Simulator Screen Shot - iPhone SE (2nd generation) - 2020-08-12 at 11 10 59](https://user-images.githubusercontent.com/1235097/90003863-a0d3f900-dc8c-11ea-8724-4319fee9ccac.png)
